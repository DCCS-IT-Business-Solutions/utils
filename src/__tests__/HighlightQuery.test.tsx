import * as React from "react";
import { shallow, mount, render } from "enzyme";
import { HighlightQuery } from "../HighlightQuery";

it("should render", () => {
  const sut = render(HighlightQuery("this is a test string", "test"));
  expect(sut).toMatchSnapshot();
});

it("should render with query being null", () => {
  const sut = render(HighlightQuery("this is a test string", null));
  expect(sut).toMatchSnapshot();
});

it("should render with text being null", () => {
  const sut = render(HighlightQuery(null, "test"));
  expect(sut).toMatchSnapshot();
});

it("should render with custom style", () => {
  const sut = render(HighlightQuery(null, "test", { backgroundColor: "red" }));
  expect(sut).toMatchSnapshot();
});

it("should highlight 'test'", () => {
  const sut = shallow(HighlightQuery("this is a test string", "test")).setProps(
    {}
  );
  const highlightedSpan = sut.find("span > span").get(0);
  const style = highlightedSpan.props.style.fontWeight;
  expect(style).toBe("bold");

  const text = sut
    .find("span > span")
    .at(0)
    .text();
  expect(text).toBe("test");
});

it("should not highlight anything", () => {
  const sut = shallow(HighlightQuery("this is a test string", "")).setProps({});

  expect(sut.children("span").length).toBe(0);
});

it("should highlight 'test' with custom style", () => {
  const sut = shallow(
    HighlightQuery("this is a test string", "test", { backgroundColor: "red" })
  ).setProps({});

  const highlightedSpan = sut.find("span > span").get(0);
  const style = highlightedSpan.props.style.backgroundColor;
  expect(style).toBe("red");
});
