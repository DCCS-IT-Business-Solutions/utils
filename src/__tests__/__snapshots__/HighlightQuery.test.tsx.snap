// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "this is a ",
        "next": Object {
          "attribs": Object {
            "style": "font-weight:bold",
          },
          "children": Array [
            Object {
              "data": "test",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "data": " string",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "text",
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
      Object {
        "attribs": Object {
          "style": "font-weight:bold",
        },
        "children": Array [
          Object {
            "data": "test",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "data": " string",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "text",
        },
        "parent": [Circular],
        "prev": Object {
          "data": "this is a ",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "style": undefined,
        },
      },
      Object {
        "data": " string",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "style": "font-weight:bold",
          },
          "children": Array [
            Object {
              "data": "test",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "data": "this is a ",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "style": undefined,
          },
        },
        "type": "text",
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`should render with custom style 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`should render with query being null 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "data": "this is a test string",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`should render with text being null 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
